config_utilities/include/config_utilities/internal/yaml_parser.h

```diff
@@ -91,6 +116,29 @@ class YamlParser {
     return error.empty();
   }
 
+  /**
+   * @brief Parse a single value to the yaml node.
+   * @param value The value to parse.
+   * @param error Optional: Where to store the error message if conversion fails.
+   */
+  template <typename T>
+  static YAML::Node toYaml(const T& value, std::string* error = nullptr) {
+    YAML::Node node;
+    std::string err;
+    try {
+      node = toYamlImpl("tmp", value, err);
+    } catch (const std::exception& e) {
+      err = std::string(e.what());
+    }
+    if (!err.empty()) {
+      if (error) {
+        *error = err;
+      }
+      return YAML::Node(YAML::NodeType::Null);
+    }
+    return node["tmp"];
+  }
+
```

Comment:
It took me a while to parse why this overload existed, mostly because of the "tmp" namespace. I think it might make more sense to have `toYamlImpl` not namespace the returned node and have the previous `toYaml` version handle namespacing

Reply:
Agreed! I've been meaning to clean this up but didn't want to touch everything, should be better now 👍